base_binary_url_import/wizard/base_binary_url_import.py

```diff
@@ -0,0 +1,302 @@
+    line_ids = fields.One2many("base.binary.url.import.line", "wizard_id")
+
+    @api.model
+    def check_access_rights(self, operation, raise_exception=True):
+        res = super().check_access_rights(operation, raise_exception=raise_exception)
+        if not self.env.user._can_import_remote_urls():
+            raise AccessError(
+                _(
+                    "You can not import files via URL, check with your "
+                    "administrator for the reason."
+                )
+            )
+        return res
+
+    def _get_ir_model_fields_domain(self, ttype):
+        self.ensure_one()
+        return [("ttype", "=", ttype), ("model_id", "=", self.target_model_id.id)]
+
+    @api.depends("target_model_id")
+    def _compute_allowed_target_fields_domain(self):
+        for wiz in self:
+            binary_domain = wiz._get_ir_model_fields_domain("binary")
+            char_domain = wiz._get_ir_model_fields_domain("char")
+            wiz.target_binary_field_domain = json.dumps(binary_domain)
+            wiz.target_binary_filename_field_domain = json.dumps(char_domain)
+
+    @api.onchange("line_ids")
+    def onchange_line_ids(self):
+        for wiz_line in self.line_ids:
+            if LINE_BREAK_CHAR in (wiz_line.binary_url_to_import or ""):
+                split_lines = wiz_line.binary_url_to_import.split(LINE_BREAK_CHAR)
+                split_lines = list(filter(None, split_lines))
+                if SEPARATOR_CHAR in split_lines[0]:
+                    line_identifier, url = split_lines[0].split(SEPARATOR_CHAR)
+                    wiz_line.update(self._prepare_wiz_line_vals(line_identifier, url))
+                wiz_lines_commands = self._generate_wiz_lines_commands(split_lines[1:])
+                self.update({"line_ids": wiz_lines_commands})
+                break
+
+        if self.line_ids:
+            self.lines_sanity_check()
+
+    def _generate_wiz_lines_commands(self, split_lines):
+        commands = []
+        for line in split_lines:
+            line_identifier, url = line.split(SEPARATOR_CHAR)
+            line_vals = self._prepare_wiz_line_vals(line_identifier, url)
+            commands.append((0, 0, line_vals))
+        return commands
+
+    def _prepare_wiz_line_vals(self, identifier, url):
+        return {
+            "target_record_identifier": identifier,
+            "binary_url_to_import": url,
+        }
+
+    def lines_sanity_check(self):
+        """Check if IDs are existing for selected model and we don't have two
+        lines matching same record"""
+        self.ensure_one()
+        db_ids = []
+        xml_ids = []
+        # Check target IDs are integers or XMLIDs
+        for wiz_line in self.line_ids:
+            id_type = wiz_line._get_identifier_type()
+            if id_type == "db_id":
+                db_ids.append(int(wiz_line.target_record_identifier))
+            elif id_type == "xml_id":
+                xml_ids.append(wiz_line.target_record_identifier)
```

Comment:
extract this part to be able to override later easily

globally, each check part of this method can be extracted so that it's easier to read/maintain/override